### PR TITLE
CV2-5191: Fix Replace by action

### PR DESCRIPTION
**Pull request by melsawy**
## Description

When user add a media do the following:

- Keep the status for the current item
- Merge assignments 
- Add a log entry for replacement

References: CV2-5191, CV2-5177

## How has this been tested?

Implemented automated tests.

## Things to pay attention to during code review

Please describe parts of the change that require extra attention during code review, for example:

- File FFFF, line LL: This refactoring does this and this. Is it consistent with how it’s implemented elsewhere?
- Etc.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have added unit and feature tests, if the PR implements a new feature or otherwise would benefit from additional testing
- [ ] I have added regression tests, if the PR fixes a bug
- [ ] I have added logging, exception reporting, and custom tracing with any additional information required for debugging
- [ ] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [ ] My changes generate no new warnings
- [ ] If I added a third party module, I included a rationale for doing so and followed our current [guidelines](https://meedan.atlassian.net/wiki/spaces/ENG/overview#Choose-the-%E2%80%9Cright%E2%80%9D-3rd-party-module)

### CODE REVIEW

File: app/models/project_media.rb [hunk 1]
```diff
@@ -295,14 +302,42 @@ def replace_by(new_pm, skip_send_report = false)
...
+        author_id: User.current&.id,
+        assignments_ids: assignments_ids,
+        skip_send_report: skip_send_report
+      }
+      self.class.delay_for(5.seconds).apply_replace_by(self.id, new_pm.id, options.to_json)
     end
   end
 
-  def self.apply_replace_by(old_pm_id, new_pm_id, skip_send_report = false)
+  def self.apply_replace_by(old_pm_id, new_pm_id, options_json)
```
Comment (caiosba) [inline app/models/project_media.rb:314]: Question: The old implementation had a default value for the `skip_send_report` argument, and now there is no default value for `options_json` - did you make sure that all replaces that call this method are passing the three arguments?

File: app/models/project_media.rb [hunk 2]
```diff
@@ -295,14 +302,42 @@ def replace_by(new_pm, skip_send_report = false)
       Rails.cache.write("check_cached_field:ProjectMedia:#{new_pm.id}:creator_name", 'Import')
 
       # Apply other stuff in background
-      self.class.delay_for(5.seconds).apply_replace_by(self.id, new_pm.id, skip_send_report)
+      options = {
+        author_id: User.current&.id,
+        assignments_ids: assignments_ids,
+        skip_send_report: skip_send_report
+      }
+      self.class.delay_for(5.seconds).apply_replace_by(self.id, new_pm.id, options.to_json)
```
Comment (caiosba) [inline app/models/project_media.rb:310]: Suggestion: Since you're here, I think this interval can be lower, right?

**Review by caiosba — APPROVED** [review]
Thanks Sawy, I left a question and a suggestion, but no blockers. Good to go once tests pass and if all new code is covered by tests.

File: app/models/project_media.rb (hunk 1 above)
Comment (melsawy) [reply app/models/project_media.rb:314]: This is the only place that calling `apply_replace_by`   but I'll make sure to run all SideKiq jobs for this one before live deployment

File: app/models/project_media.rb (hunk 2 above)
Comment (melsawy) [reply app/models/project_media.rb:310]: right

**Pull request merged**